### ISSUE TITLE
[request] Add pretty json support

### ISSUE THREAD

**Issue by masbug** [niXman/yas#156]
For debugging etc. it's really usefull to have pretty printed json.

**Comment by ambitslix** [comment]
You can use rapidjson. I think it's outside the scope of yas.

**Comment by masbug** [comment]
I'm pretty sure it's not. I really like YAS because it is very fast and easy to add to the code. I would hate to use another json lib just to add pretty printing.
To add printing a "size_t level" would need to be added to the serializer template...after that it's easy ;)

**Comment by ambitslix** [comment]
I would agree but if you need json anything you will need a c++ library. You are not just going to use yas for your c++ json business? How are you going to parse json for example?
